### DISCUSSION TITLE
Welcome to TEAMMATES Discussions!

### DISCUSSION THREAD

**Discussion by madanalogy** [TEAMMATES/teammates#11055] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by retsofoster** [comment]
Hello, I am new to open source projects and I am attempting to set up a development environment on my local machine using the [development guidelines](https://teammates.github.io/teammates/setting-up.html) link. I am having trouble at the "npm run start step" as it says there are multiple errors and fails to compile as you can see in the attached picture. Any help or guidance would be appreciated. 
<img width="1792" alt="Screen Shot 2023-02-11 at 11 21 13 PM" src="https://user-images.githubusercontent.com/104222559/218292845-5083b6da-f8f9-4c28-b9e1-fee6e1589aa9.png">

**Reply by zhaojj2209** [reply]
Thank you for your interest in TEAMMATES!

Please follow the development guidelines section on [starting the frontend dev server](https://teammates.github.io/teammates/development.html#managing-the-dev-server-front-end) to compile type definitions from the backend first before starting the frontend.

**Reply by retsofoster** [reply]
Thank you for such a quick reply, I actually did make it to this page and the first command runs fine its the second command that I get stuck on.

**Reply by zhaojj2209** [reply]
Can you try deleting the `node_modules` folder and reinstalling the dependencies as per the [set-up guide](https://teammates.github.io/teammates/setting-up.html)?

**Reply by retsofoster** [reply]
Thank you that worked!

**Reply by retsofoster** [reply]
Ok, so it seems as if I have gotten docker to run my teammates datastore and run the front end on port 4200. Although when I load the cite http://localhost:4200 on my browser nothing is displayed. Your help is always appreciated.
<img width="1792" alt="Screen Shot 2023-02-12 at 4 33 18 PM" src="https://user-images.githubusercontent.com/104222559/218338452-b95b3e35-65c7-450e-a473-5c5920462e65.png">

**Reply by zhaojj2209** [reply]
http://localhost:4200 should redirect to http://localhost:4200/web/front/home. Please try refreshing the page or visiting http://localhost:4200/web/front/home directly.

Hiding this conversation as the majority of issues have been resolved. If you have any other queries, please open an issue instead.

**Comment by samuelfangjw** [comment]
Locking this discussion to keep things neat. Please feel free to open a new discussion under the appropriate category if you have any questions or ideas!
